packages/jaeger-ui/src/utils/TreeNode.js

```diff
@@ -25,8 +25,9 @@ export default class TreeNode {
     return (value, node) => (search instanceof TreeNode ? node === search : value === search);
   }
 
-  constructor(value, children = []) {
+  constructor(value, endTime = Infinity, children = []) {
```

Comment:
This looks to me breaking the existing API for other functions. In general, doing any code refactor, which changes the API contract/function signature/expectations, should be addressed in a different PR.

I would rather use another way to save the information that you are storing in this object. Maybe have an external `Map[NodeID, endTime]` should help you solve your problem?

Reply:
As there is already a ``Map[SpanId,Span]`` I reused it .I updated code like this Is it fine?? Without changing TreeNode structure
https://github.com/jaegertracing/jaeger-ui/blob/1be78649fea87686494f2651d1b67fdb49821f3f/packages/jaeger-ui/src/model/transform-trace-data.tsx#L138-L146

Reply:
The changes make sense to me.